### ISSUE TITLE
Documentation Issue

### ISSUE THREAD

**Issue by XDapps** [rudreshsolanki97/xdc-connect#7]
Hello, this is not really a bug, but I wasn't sure where else to ask this question. I can successfully connect to to the network and read balances, but I cannot find any docs on these two methods:

SendTransaction - returns promise which will resolve into xdc3 receipt for the transaction.
CallTransaction - returns promise which will resolve into response for the transaction.

What are the arguments that need to be passed to these functions?

In Ethers.js I would do it like this below, how do I perform the same in xdc-connect?

Read Only Call:
```
 const getTokenURI = async (_tokenId: number): Promise<string> => {
    const abi = ["function tokenURI(uint256 tokenId) public view returns (string memory)"];
    try {
      const tokenContract = new ethers.Contract(app.vars.tokenContractAddress, abi, ethersProvider);
      return await tokenContract.tokenURI(_tokenId);
    } catch (error) {
      console.log('Error ', error)
      return error;
    }
  }

  ```

Send Transaction with Data Call:
```  
const approveTokenOperator = async (_tokenId: number): Promise<void> => {    
const abi = ["function approve(address to, uint256 tokenId) public"];
    try {
      const tokenContract = new ethers.Contract(contractAddress, abi, ethersProvider.getSigner());
      return await tokenContract.approve(operatorAddress, _tokenId);
    } catch (error) {
      console.log('Error ', error)
      return error;
    }
  }
```

![xdc connect1](https://user-images.githubusercontent.com/89046722/130331248-214622d9-3979-46c4-a4e9-221e85fd2a16.png)

Thank you in advance!

**Comment by rudreshsolanki97** [comment]
Hello, it accepts a TX object as per the interface specified at https://web3js.readthedocs.io/en/v1.2.2/web3-eth.html#id71.

Thanks for bringing this up, we will add this to documentation as well. I Will close this once I have added the required documentation.

**Comment by XDapps** [comment]
I got this working, but since there are no types available on xdc-connect for Typescript I have to split into two files until we can get the type files.

xdc3Helpers.ts
```
import { readContract } from './xdc3CallTx';
import { balanceOfABI } from '../../models/abi/abi';
import xdc3 from 'xdc3';
const x3 = new xdc3();

export const getBalanceOf = async (addressToCheck: string) => {
  const functionCallEncoded = getEncodedFunctionCall(balanceOfABI, [addressToCheck]);
  const tx = {
    to:<toAddress>,
    from:<fromAddress>,
    data: functionCallEncoded
  }
  const txResult = await readContract(tx);
  console.log('CallTransaction ', txResult);
}



const getEncodedFunctionCall = (functionABI: string, paramData: any[]) => {
  const parsedABI = JSON.parse(functionABI);
  return x3.eth.abi.encodeFunctionCall(parsedABI, paramData);
}

```

And then xdc3CallTx.js

```
import { CallTransaction } from "xdc-connect";
export const readContract = async (tx) => {
   return await CallTransaction(tx);
}
```

**Comment by XDapps** [comment]
I'm having another minor issue. 

Calling a read function through Remix returns the expected result of string: "QmdPSRYhpiYdXqKwmh95Lbe9dS6Z4V87BXzpPPPdczuvEY"

Calling the same read function via the format above returns:
"0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002e516d6450535259687069596458714b776d6839354c6265396453365a3456383742587a7050505064637a75764559000000000000000000000000000000000000"

I'm guessing my expected result is somehow encoded in that result, but I have not figured out how to decode this properly yet.

**Comment by rudreshsolanki97** [comment]
can you let me know the contract abi, address & the function you're calling ?

**Comment by XDapps** [comment]
> can you let me know the contract abi, address & the function you're calling ?

Yes, thank you! The contract is on main net at 0x50d366a72012dfddae856e5e4525e8d01b698560.

I am trying to call the tokenURI function with the number 1 passed as the tokenId param. I am expecting the result received from Remix of "QmdPSRYhpiYdXqKwmh95Lbe9dS6Z4V87BXzpPPPdczuvEY".

I'm sure there is a decoding step I am missing somewhere.

The function ABI is 

```

     { "constant": true, "inputs": [ 
     { "name": "tokenId", "type": "uint256" } ], "name": "tokenURI", "outputs": [
     { "name": "", "type": "string" } ], "payable": false, "stateMutability": "view", "type": "function"
      }
                                         
     
```

**Comment by rudreshsolanki97** [comment]
You are correct. The data returned is an encoded hex, you'll need to decode the hex to string.
Please envoke the following function on the encoded HEX string.  

```
const decoded = xdc3.eth.abi.decodeParameter("string", encodedHex); // here "string" is the type of parameter
```

**Comment by XDapps** [comment]
> You are correct. The data returned is an encoded hex, you'll need to decode the hex to string.
> Please envoke the following function on the encoded HEX string.
> 
> ```
> const decoded = xdc3.eth.abi.decodeParameter("string", encodedHex); // here "string" is the type of parameter
> ```

This worked perfectly and is exactly what I needed, thank you so much.

When I am finished with this project, I'll try to put together a summary of best practices from my findings. You're welcome to close this issue.

Thank you again!

**Comment by rudreshsolanki97** [comment]
closed  in 527e3fd
